pkg/reconciler/cache/cachedresourceendpointslice/cachedresourceendpointslice_controller.go

```diff
@@ -0,0 +1,250 @@
+	cachev1alpha1informers "github.com/kcp-dev/kcp/sdk/client/informers/externalversions/cache/v1alpha1"
+	corev1alpha1informers "github.com/kcp-dev/kcp/sdk/client/informers/externalversions/core/v1alpha1"
+)
+
+const (
+	ControllerName = "kcp-cachedresourceendpointslice"
+)
+
+// NewController returns a new controller for CachedResourceEndpointSlices.
+func NewController(
+	shardName string,
+	cachedResourceEndpointSliceClusterInformer cachev1alpha1informers.CachedResourceEndpointSliceClusterInformer,
+	cachedResourceClusterInformer cachev1alpha1informers.CachedResourceClusterInformer,
+	globalShardClusterInformer corev1alpha1informers.ShardClusterInformer,
+	lcClusterInformer corev1alpha1informers.LogicalClusterClusterInformer,
+	apiBindingClusterInformer apisv1alpha2informers.APIBindingClusterInformer,
+	kcpClusterClient kcpclientset.ClusterInterface,
+) (*controller, error) {
+	c := &controller{
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
+			workqueue.DefaultTypedControllerRateLimiter[string](),
+			workqueue.TypedRateLimitingQueueConfig[string]{
+				Name: ControllerName,
+			},
+		),
+		listCachedResourceEndpointSlices: func() ([]*cachev1alpha1.CachedResourceEndpointSlice, error) {
+			return cachedResourceEndpointSliceClusterInformer.Lister().List(labels.Everything())
+		},
+		getCachedResourceEndpointSlice: func(clusterName logicalcluster.Name, name string) (*cachev1alpha1.CachedResourceEndpointSlice, error) {
+			return cachedResourceEndpointSliceClusterInformer.Cluster(clusterName).Lister().Get(name)
+		},
+		getCachedResource: func(clusterName logicalcluster.Name, name string) (*cachev1alpha1.CachedResource, error) {
+			return cachedResourceClusterInformer.Cluster(clusterName).Lister().Get(name)
+		},
+		getMyShard: func() (*corev1alpha1.Shard, error) {
+			return globalShardClusterInformer.Cluster(core.RootCluster).Lister().Get(shardName)
+		},
+		getLogicalCluster: func(clusterName logicalcluster.Name) (*corev1alpha1.LogicalCluster, error) {
+			return lcClusterInformer.Cluster(clusterName).Lister().Get("cluster")
+		},
+		getAPIBinding: func(clusterName logicalcluster.Name, bindingName string) (*apisv1alpha2.APIBinding, error) {
+			return apiBindingClusterInformer.Cluster(clusterName).Lister().Get(bindingName)
+		},
+		cachedResourceEndpointSliceClusterInformer: cachedResourceEndpointSliceClusterInformer,
+		commit: committer.NewCommitter[*CachedResourceEndpointSlice, Patcher, *CachedResourceEndpointSliceSpec, *CachedResourceEndpointSliceStatus](kcpClusterClient.CacheV1alpha1().CachedResourceEndpointSlices()),
+	}
+
+	logger := logging.WithReconciler(klog.Background(), ControllerName)
+
+	_, _ = cachedResourceEndpointSliceClusterInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			c.enqueueCachedResourceEndpointSlice(objOrTombstone[*cachev1alpha1.CachedResourceEndpointSlice](obj), logger, "")
```

Comment:
Drop the logsuffix if we dont use it. Usually we use it where are multipe enqueue entrypoints and we want to note reason for it.